### PR TITLE
Fix NPE

### PR DESCRIPTION
**Pull request by rubenporras**
for https://github.com/eclipse/lsp4e/issues/492

### CODE REVIEW

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServers.java [hunk 1]
```diff
@@ -458,5 +463,6 @@ private <T> void completeEmptyOrWithException(final CompletableFuture<Optional<T
 
 	private @NonNull Predicate<ServerCapabilities> filter = s -> true;
 
-	protected @Nullable Comparator<? super LanguageServerWrapper> comparator;
+	protected @Nullable String serverId;
+;
```
Comment (mickaelistria) [inline org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServers.java:467]:
```suggestion

```

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServers.java [hunk 2]
```diff
@@ -164,11 +164,12 @@ public <T> CompletableFuture<Optional<T>> computeFirst(BiFunction<? super Langua
 		return result.thenApplyAsync(Function.identity());
 	}
 
-	public @NonNull E withComparator(final @Nullable Comparator<? super LanguageServerWrapper> comparator) {
-		this.comparator = comparator;
+	public @NonNull E withPreferredServer(final @Nullable String serverId) {
```
Comment (mickaelistria) [inline org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServers.java:167]: Could we use a LanguageServerDefiniton here? Stronger typing usually makes code less error-prone.

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServers.java (hunk 1 above)
Comment (rubenporras) [reply org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServers.java]: thanks.

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServers.java (hunk 2 above)
Comment (rubenporras) [reply org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServers.java:167]: the consumer does not have the `LanguageServerDefiniton`, it only has the id, so I think that for this PR is it fine to leave it as this.

Comment (mickaelistria) [reply org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServers.java:167]: The consumer can get the languageServerDefinition from the id (`LanguageServerRegistry.getDefinition(id)`). Really, one never regrets stronger typing.

Comment (rubenporras) [reply org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServers.java]: Ok, done

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/diagnostics/LSPDiagnosticsToMarkers.java [hunk 3]
```diff
@@ -207,7 +216,7 @@ private IMarker getExistingMarkerFor(IDocument document, Diagnostic diagnostic,
 				if (LSPEclipseUtils.toOffset(diagnostic.getRange().getStart(), document) == MarkerUtilities.getCharStart(marker)
 						&& LSPEclipseUtils.toOffset(diagnostic.getRange().getEnd(), document) == MarkerUtilities.getCharEnd(marker)
 						&& Objects.equals(marker.getAttribute(IMarker.MESSAGE), diagnostic.getMessage())
-						&& Objects.equals(marker.getAttribute(LANGUAGE_SERVER_ID), this.languageServerId)) {
+						&& Objects.equals(marker.getAttribute(LANGUAGE_SERVER_ID), languageServerDefinition)) {
```
Comment (mickaelistria) [inline org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/diagnostics/LSPDiagnosticsToMarkers.java:219]: I believe marker do require Serializable objects of something of that sort as they are persisted in the workspace. If so, we'll need to stick to id.

Comment (rubenporras) [reply org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/diagnostics/LSPDiagnosticsToMarkers.java]: Okay, then back to the roots. If it passes the tests I will merge it. We can always improve later if wanted

**Pull request merged**